### PR TITLE
Mt Pelerin offramp integration

### PR DESCRIPTION
**Pull request by mateodaza**
This leaves the code of onramper + donorbox hidden but ready for later 

This PR replaces [this one](https://github.com/Giveth/giveth-dapps-v2/pull/1872)

### CODE REVIEW

File: src/components/views/purchase/HowToBuyXdai.tsx [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+			id: 8,
+			title: 'label.confirmation',
+			description: 'label.wait_for_confirmation',
+		},
+		{
+			id: 9,
+			title: 'label.transaction_status',
+			description: 'label.if_the_payment_is_approved_it_will_display',
+		},
+	];
```
Comment (MohammadPCh) [inline src/components/views/purchase/HowToBuyXdai.tsx:93]: It's better to move it to the out of the component.

File: src/components/views/purchase/PurchaseXDai.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+								{formatMessage({
+									id: 'label.kyc_less_the_service_is_based_in_sw',
+								})}
+							</P>
+						</Info>
+						<PwdMtPelerin>
+							Powered By <img src='/images/mtpelerin.svg' />
+						</PwdMtPelerin>
+					</Left>
+					<Right isMobile={isMobile}>
```
Comment (MohammadPCh) [inline src/components/views/purchase/PurchaseXDai.tsx:54]: It's better to handle this with media queries.

File: src/types/common.ts [hunk 3]
```diff
@@ -1,3 +1,5 @@
 export interface IModal {
 	setShowModal: (showModal: boolean) => void;
+	continueProcess?: () => void;
+	type?: string;
```
Comment (MohammadPCh) [inline src/types/common.ts:4]: please add these to your specific interface, we should keep the base interface abstract.

**Review by MohammadPCh — CHANGES_REQUESTED** [review]
@mateodaza Thanks, I added some suggestions.

File: .github/workflows/Build.yml [hunk 4]
```diff
@@ -6,12 +6,11 @@ jobs:
...
-            - uses: actions/checkout@v3
+            - uses: actions/checkout@v1
             - name: Use Node.js
-              uses: actions/setup-node@v3
+              uses: actions/setup-node@v1
               with:
-                  node-version: '18'
-                  check-latest: true
+                  node-version: 16.x
             - name: Install dependencies
```
[the remark this answers is not in the document]
Comment (mateodaza) [reply .github/workflows/Build.yml:14]: the onramper widget won't work with v18 😢 
I can leave it like this for the mt pelerin version and worry about this with them later - maybe ask them to upgrade it

File: package.json [hunk 5]
```diff
@@ -11,6 +11,9 @@
 		"test": "jest --watch",
 		"cypress": "cypress open"
 	},
+	"engines": {
+		"node": ">=16.11.1"
+	},
```
[the remark this answers is not in the document]
Comment (mateodaza) [reply package.json:16]: same reason as above, will fix

**Review by MohammadPCh — APPROVED** [review]
LGTM, Thanks @mateodaza

**Pull request merged**